debian/clearwater-socket-factory.prerm

```diff
@@ -52,6 +52,12 @@ set -e
 
 case "$1" in
     remove|upgrade|deconfigure)
+        # Stop and deconfigure the management and signaling upstart jobs
+        stop clearwater-socket-factory-mgmt > /dev/null 2>&1 || true
+        stop clearwater-socket-factory-sig > /dev/null 2>&1 || true
```

Comment:
I've checked https://www.debian.org/doc/debian-policy/ch-maintainerscripts.html and during an upgrade the prerm is called before the postinst. So I'm happy with your prerm script. However I don;t think you need the equivilent lines at the start of your postinst